python/lsst/ts/standardscripts/maintel/m1m3/check_actuators.py

```diff
@@ -0,0 +1,261 @@
+                    items:
+                      type: number
+                      enum: [{m1m3_actuator_ids_str}]
+                    minItems: 1
+                    uniqueItems: true
+                    additionalItems: false
+                  - type: string
+                    enum: ["all"]
+                default: "all"
+        additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    async def configure(self, config):
+        """Configure the script.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+            Configuration
+        """
+
+        # Getting actuators to be tested.
+        self.actuators_to_test = (
+            self.m1m3_actuator_ids if config.actuators == "all" else config.actuators
+        )
+
+    def set_metadata(self, metadata):
+        """Set metadata."""
+
+        # Getting total number of secondary actuators to be tested
+        total_tested_secondary = sum(
+            [
+                1
+                for actuator in self.actuators_to_test
+                if self.has_secondary_actuator(actuator)
+            ]
+        )
+
+        # Setting metadata
+        metadata.duration = self.time_one_bump * (
+            len(self.actuators_to_test) + total_tested_secondary
+        )
+
+    def has_secondary_actuator(self, actuator_id: int) -> bool:
+        """Determines whether a given actuator has a
+        secondary axis or not.
+        """
+
+        return actuator_id in self.m1m3_secondary_actuator_ids
+
+    async def run(self):
+        await self.assert_feasibility()
+        start_time = time.time()
+
+        # Get M1M3 detailed state
+        detailed_state = DetailedState(
+            (
+                await self.mtcs.rem.mtm1m3.evt_detailedState.aget(
+                    timeout=self.mtcs.fast_timeout,
+                )
+            ).detailedState
+        )
+        self.log.info(f"Current M1M3 detailed state: {detailed_state!r}.")
+
+        # Put M1M3 in engineering mode
+        await self.mtcs.enter_m1m3_engineering_mode()
+
+        failed_actuators = []
+        for i, actuator_id in enumerate(self.actuators_to_test):
+            await self.checkpoint(f"Running bump test on FA ID: {actuator_id}.")
+
+            # Determine if actuator has a secondary actuator
+            secondary_exist = self.has_secondary_actuator(actuator_id)
+
+            try:
+                await self.mtcs.run_m1m3_actuator_bump_test(
+                    actuator_id=actuator_id,
+                    primary=True,
+                    secondary=secondary_exist,
+                )
+            # raise a RuntimeError if the bump test fails
+            # and continue loop to next actuator_id
+            except RuntimeError as e:
+                failed_actuators.append(actuator_id)
+
+                self.log.error(
+                    f"Failed to run bump test on FA ID {actuator_id}." f"Error: {e}"
```

Comment:
you can merge the 2 strings into 1 apparently:

```
f"Failed to run bump test on FA ID {actuator_id}. Error: {e}"
```

However, if you want to log the exception, use `self.log.exception` instead:

```
except RuntimeError:
    failed_actuators.append(actuator_id)
    self.log.exception(f"Failed to run bump test on FA ID {actuator_id}.")
```